### PR TITLE
MM-47083 : Migrate "components/pdf_preview.jsx" and tests to Typescript

### PR DESCRIPTION
**Pull request by jatin33**
#### Summary

Make pdf review component typescript aware

#### Ticket Link

JIRA: https://mattermost.atlassian.net/browse/MM-47083
Fixes https://github.com/mattermost/mattermost/issues/21086

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE

-->
```release-note
None
```

### CODE REVIEW

File: webapp/channels/src/components/pdf_preview.tsx [hunk 1]
```diff
@@ -96,25 +120,25 @@ export default class PDFPreview extends React.PureComponent {
...
-    downloadFile = (e) => {
-        const fileDownloadUrl = this.props.fileInfo.link || getFileDownloadUrl(this.props.fileInfo.id);
+    downloadFile = (e: any) => {
+        const fileDownloadUrl = this.props.fileInfo.link || getFileDownloadUrl(this.props.fileInfo.id || '');
         e.preventDefault();
         window.location.href = fileDownloadUrl;
     };
 
-    isInViewport = (page) => {
+    isInViewport = (page: any) => {
```
Comment (larkox) [inline webapp/channels/src/components/pdf_preview.tsx:129]: Can we get the correct type for page?

File: webapp/channels/src/components/pdf_preview.tsx [hunk 2]
```diff
@@ -96,25 +120,25 @@ export default class PDFPreview extends React.PureComponent {
...
         const bounding = page.getBoundingClientRect();
-        const viewportTop = this.container.current.scrollTop;
-        const viewportBottom = viewportTop + this.container.current.parentElement.clientHeight;
+        const viewportTop: any = this.container?.current?.scrollTop;
```
Comment (larkox) [inline webapp/channels/src/components/pdf_preview.tsx:131]: Can we better type this?

File: webapp/channels/src/components/pdf_preview.tsx [hunk 3]
```diff
@@ -215,7 +239,7 @@ export default class PDFPreview extends React.PureComponent {
         if (!this.state.success) {
             return (
                 <FileInfoPreview
-                    fileInfo={this.props.fileInfo}
+                    fileInfo={this.props.fileInfo as FileInfo}
```
Comment (larkox) [inline webapp/channels/src/components/pdf_preview.tsx:242]: This should already be a FileInfo. Why is the casting needed?

**Review by larkox — CHANGES_REQUESTED** [review]
Some changes needed regarding typing.

File: webapp/channels/src/components/pdf_preview.tsx (hunk 3 above)
Comment (jatin33) [reply webapp/channels/src/components/pdf_preview.tsx]: fixed

File: webapp/channels/src/components/pdf_preview.tsx (hunk 2 above)
Comment (jatin33) [reply webapp/channels/src/components/pdf_preview.tsx]: fixed

File: webapp/channels/src/components/pdf_preview.tsx (hunk 1 above)
Comment (jatin33) [reply webapp/channels/src/components/pdf_preview.tsx]: fixed

File: webapp/channels/src/components/pdf_preview.tsx [hunk 4]
```diff
@@ -96,25 +120,25 @@ export default class PDFPreview extends React.PureComponent {
         }
     }
 
-    downloadFile = (e) => {
-        const fileDownloadUrl = this.props.fileInfo.link || getFileDownloadUrl(this.props.fileInfo.id);
+    downloadFile = (e: any) => {
```
[the remark this answers is not in the document]
Comment (jatin33) [reply webapp/channels/src/components/pdf_preview.tsx]: fixed

File: webapp/channels/src/components/pdf_preview.tsx [hunk 5]
```diff
@@ -30,11 +53,12 @@ export default class PDFPreview extends React.PureComponent {
         handleBgClose: PropTypes.func.isRequired,
     };
 
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
 
         this.pdfPagesRendered = {};
         this.container = React.createRef();
+        this.pdfCanvasRefs = [];
```
[the remark this answers is not in the document]
Comment (jatin33) [reply webapp/channels/src/components/pdf_preview.tsx]: fixed

File: webapp/channels/src/components/file_preview_modal/file_preview_modal.tsx [hunk 6]
```diff
@@ -28,7 +28,7 @@ import FilePreviewModalHeader from './file_preview_modal_header/file_preview_mod
 import PopoverBar from './popover_bar';
 import {LinkInfo, isFileInfo} from './types';
 
-const PDFPreview = React.lazy(() => import('components/pdf_preview'));
+const PDFPreview = React.lazy(() => import('components/pdf_preview') as any);
```
[the remark this answers is not in the document]
Comment (jatin33) [reply webapp/channels/src/components/file_preview_modal/file_preview_modal.tsx]: fixed

**Comment by M-ZubairAhmed** [comment]
Gentle ping on this one @larkox 😊

**Review by larkox — APPROVED** [review]
LGTM. Thank you!

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Please update your branch to latest master and check the failing CI

**Comment by M-ZubairAhmed** [comment]
@jatin33 apologies for delay on review on this from myside, can we please get this to finish line

**Comment by jatin33** [comment]
@M-ZubairAhmed sorry for the delay in response, been travelling. I will be available after 10 days and will wrap this up then

**Comment by M-ZubairAhmed** [comment]
@jatin33 can we please get this to finish line

**Comment by M-ZubairAhmed** [comment]
@jatin33 do you plan to finish this off?

**Pull request closed without merging**